test/canvas.spec.js

```diff
@@ -1,35 +1,58 @@
-// const http = require('http');
-// process.env.NODE_ENV = 'test';
-// const app = require('../src/server');
-// const idea = require('../public/javascript/idea.js');
-// const Browser = require('zombie');
-// const chai = require('chai');
-// const expect = chai.expect;
-//
-// describe('User sees blank canvas', function() {
-//
-//   before(function() {
-//     this.server = http.createServer(app).listen(8080);
-//     this.browser = new Browser({ site: 'http://localhost:8080'});
-//   });
-//
-//   before (function(done) {
-//     this.browser.visit('/index', done);
-//   });
-//
-//   it('should have a blank canvas', function() {
-//     this.browser.assert.element('#canvas');
-//   });
-//
-//   it('should show an addded object', function() {
-//     canvasObject = new Idea(document.getElementById('canvas').getContext('2d'));
-//     // canvasObject.context.fillStyle = "cornflowerblue"
-//     this.browser.assert.element('#canvas1')
-//     // canvasObject.context.clear();
-//   });
-//
-//   after(function(done) {
-//     this.server.close(done);
-//   });
-//
-// });
+const app = require('../src/server');
+const LineFile = require('../public/./javascript/canvas.js');
+Browser = require('zombie');
+let assert = require('chai').assert;
+describe('Canvas unit/feature tests', function() {
+
+  before(function() {
+    this.server = app.listen(8000);
+    this.browser = new Browser({ site: 'http://localhost:8000'});
+  });
+
+  before(function(done) {
+    this.browser.visit('/', function() {
+      done();
+    });
+  });
+
+  after(function(done) {
+    this.server.close(done);
+  });
+
+  beforeEach(function() {
+    line = new Line(document.getElementById('canvas').getContext('2d'));
+  });
+
+  xit('has ideas that are added to the canvas using the rectangle or circle button', function() {
+    // unit test unfinished
+  });
+
+  xit('has ideas that can be dragged to different locations on the canvas using a mouse click', function() {
+    // unit test unfinished
+  });
+
+  xit('starts to draw a line on the canvas on a double click' , function() {
+    // unit test unfinished
+  });
+
+  xit('finishes drawing the line on the canvas on a second double click' , function() {
+    // unit test unfinished
+  });
+
+  xit('finishes drawing the line on the canvas on a second double click' , function() {
+    // unit test unfinished
+  });
+
+  xit('is cleared if the user clicks on the reset button' , function() {
+    // unit test unfinished
+  });
+
+  xit('is cleared if the user clicks on the reset button' , function() {
+    // unit test unfinished
+  });
+
+  xit('can have all of the lines cleared if the user clicks on the clear lines button' , function() {
+    // unit test unfinished
+  });
+
+});
```

Comment:
I really like that you've added a road map of future tests here, I think this is 100% the best way of balancing time constraints and showing what we would have liked to do.